qa/tasks/cephfs/test_volumes.py

```diff
@@ -733,6 +741,48 @@ def test_subvolume_resize_infinite_size_future_writes(self):
             raise RuntimeError("expected filling subvolume {0} with {1} file of size {2}MB "
                                "to succeed".format(subvolname, number_of_files, file_size))
 
+    def test_subvolume_info(self):
+        # tests the 'fs subvolume info' command
+
+        subvol_md = ["subvolpath", "bytes_quota", "bytes_pcent", "data_pool", "createdAt"]
+
+        # create subvolume
+        subvolume = self._generate_random_subvolume_name()
+        self._fs_cmd("subvolume", "create", self.volname, subvolume)
+
+        # get subvolume metadata
+        subvol_info = json.loads(self._get_subvolume_info(self.volname, subvolume))
+        if len(subvol_info) == 0:
+            raise RuntimeError("Expected the 'fs subvolume info' command to list metadata of subvolume")
+        else:
+            for md in subvol_md:
+                if md not in subvol_info.keys():
+                    raise RuntimeError("%s not present in the metadata of subvolume" % md)
+
+            if subvol_info["bytes_pcent"] != "no-limit":
+                raise RuntimeError("bytes_pcent should be set to no-limit if quota is not set")
+
+            if subvol_info["bytes_quota"] != "no-limit":
+                raise RuntimeError("bytes_quota should be set to no-limit if quota is not set")
+
+        nsize = self.DEFAULT_FILE_SIZE*1024*1024
+        try:
+            self._fs_cmd("subvolume", "resize", self.volname, subvolume, str(nsize))
+        except CommandFailedError:
+            raise RuntimeError("expected the 'fs subvolume resize' command to succeed")
+
+        # get subvolume metadata after quota set
+        subvol_info = json.loads(self._get_subvolume_info(self.volname, subvolume))
+        if len(subvol_info) == 0:
+            raise RuntimeError("Expected the 'fs subvolume info' command to list metadata of subvolume")
+        else:
+            if subvol_info["bytes_pcent"] == "no-limit":
+                raise RuntimeError("bytes_pcent should not be set to no-limit if quota is set")
+
+            if subvol_info["bytes_quota"] == "no-limit":
+                raise RuntimeError("bytes_quota should not be set to no-limit if quota is set")
+
```

Comment:
also, remove the subvolume and wait till it gets async purged.

Reply:
Done